client/me/purchases/add-credit-card/index.jsx

```diff
@@ -21,13 +21,19 @@ import titles from 'me/purchases/titles';
 import { billingHistory } from 'me/purchases/paths';
 import PageViewTracker from 'lib/analytics/page-view-tracker';
 import { StripeHookProvider } from 'lib/stripe';
+import { getCurrentUserId } from 'state/current-user/selectors';
+import { isUserPaid } from 'state/purchases/selectors';
 
 function AddCreditCard( props ) {
 	const createAddCardToken = ( ...args ) => createCardToken( 'card_add', ...args );
 	const goToBillingHistory = () => page( billingHistory );
 	const recordFormSubmitEvent = () =>
 		analytics.tracks.recordEvent( 'calypso_add_credit_card_form_submit' );
 
+	if ( ! props.isUserPaid ) {
```

Comment:
This is maybe more semantic than functional, but I think this should be wrapped in a `useEffect` since it doesn't really need to happen multiple times:

```js
useEffect(() => {
  if (!props.isUserPaid) {
		page.redirect( '/me/purchases' );
  }
}, [props.isUserPaid]);
```